pipeline/container/frontend/index.html

```diff
@@ -1022,32 +1044,55 @@
           );
         });
       }
+      async function handlePostFile(file) {
+        const formData = new FormData();
+        formData.append("pfile", file);
+        const fileData = await fetch(`${host}/v4/files`, {
+          method: "POST",
+          headers: {
+            Accept: "/*",
+          },
+          body: formData,
+        }).then((res) => res.json());
+        return fileData;
+      }
 
-      function generateInputDataArrayFromDynamicFields(dynamicFields) {
+      async function generateInputDataArrayFromDynamicFields(dynamicFields) {
         const array = [];
-        dynamicFields.forEach((item) => {
+        for (const item of dynamicFields) {
           if (item.dicts && item.dicts.length) {
             const newObj = {};
-            for (const key in item.dicts) {
-              newObj[item.dicts[key].label] = item.dicts[key].defaultValue;
+            for (const dict of item.dicts) {
+              newObj[dict.label] = dict.defaultValue;
+              if (dict.subType === "file") {
+                const fileData = await handlePostFile(dict.defaultValue);
+                newObj[dict.label] = {
+                  type: "file",
+                  value: null,
+                  file_path: fileData.path,
+                };
+              }
             }
-
             array.push({
               type: item.subType,
               value: newObj,
             });
           } else {
             array.push({ type: item.subType, value: item.defaultValue });
+            if (item.subType === "file") {
+              const fileData = await handlePostFile(item.defaultValue);
+              array[array.length - 1].value = null;
+              array[array.length - 1].file_path = fileData.path;
+            }
           }
-        });
+        }
 
         return array;
       }
 
       function App() {
         const [dynamicFields, setDynamicFields] = React.useState([]);
         const [isLoading, setIsLoading] = React.useState(false);
-        const [inputData, setInputData] = React.useState([]);
```

Comment:
I'm confused, how are we tracking this info now?

Reply:
Yeah so have changed it so that inputData is only constructed when the [form gets submitted](https://github.com/mystic-ai/pipeline/blob/plutopulp/frontend-file-inputs/pipeline/container/frontend/index.html#L1300-L1316). The dynamicFields, like before, track changes in the form state, so we only need to construct the inputData from the dynamicFields once really. This also has the benefit that files will only be uploaded when submitting the form, rather than each time a file is selected.

Reply:
I see! good optimisation. LGTM.